.github/workflows/ci.yaml

```diff
@@ -22,11 +22,17 @@ concurrency:
 
 jobs:
   test:
-    name: "Unit Tests and Type Checking"
+    name: "${{ matrix.os }} :: ${{ matrix.version }} :: Unit Tests and Type Checking"
```

Comment:
@marbre you may have opinions on the job naming. I'm not sure about this.

BTW, type checking does make sense to run on multiple python versions. The use of `logging.getLevelNamesMapping()  # Added in 3.11` was caught by running mypy on 3.10. I originally wanted to move type checking to pre-commit since it's similar to other lint checks, but I now think it makes sense to keep it here

Reply:
Following the naming in other repos we would have `"Unit Tests and Type Checking :: ${{ matrix.os }} :: ${{ matrix.version }}"` but to be honest I don't really care and if we start to dislike we can change it in the future. Thus I don't want to be picky here :wink:

Reply:
Yeah I'd like to put important information at the start of the name so it doesn't get cut off

* ![image](https://github.com/user-attachments/assets/86cacd7f-2e73-4f77-99df-9e45d13b00c5)
* ![image](https://github.com/user-attachments/assets/ab0fc37b-b3f9-4f2b-93e5-297c8a2183b2)

Reply:
Works for me :)